File: src/test/scala/gameover/fwk/gdx/FakeGL20.scala
package gameover.fwk.gdx

import java.nio.{FloatBuffer, Buffer, IntBuffer}

import com.badlogic.gdx.graphics.GL20

/**
 * Fake GL for test.
 */
class FakeGL20 extends GL20 {
  override def glUseProgram(program: Int){}

  override def glCreateProgram(): Int = 0

  override def glDisableVertexAttribArray(index: Int){}

  override def glGenerateMipmap(target: Int) {}

  override def glUniform1i(location: Int, x: Int) {}

  override def glGetAttachedShaders(program: Int, maxcount: Int, count: Buffer, shaders: IntBuffer){}

  override def glGenBuffer(): Int = 0

  override def glFramebufferRenderbuffer(target: Int, attachment: Int, renderbuffertarget: Int, renderbuffer: Int){}

  override def glUniform3fv(location: Int, count: Int, v: FloatBuffer){}

  override def glUniform3fv(location: Int, count: Int, v: Array[Float], offset: Int){}

  override def glGetUniformfv(program: Int, location: Int, params: FloatBuffer){}

  override def glIsBuffer(buffer: Int): Boolean = true

  override def glDepthFunc(func: Int){}

  override def glGetShaderPrecisionFormat(shadertype: Int, precisiontype: Int, range: IntBuffer, precision: IntBuffer){}

  override def glGetAttribLocation(program: Int, name: String): Int = 0

  override def glValidateProgram(program: Int){}

  override def glGetTexParameteriv(target: Int, pname: Int, params: IntBuffer){}

  override def glStencilFuncSeparate(face: Int, func: Int, ref: Int, mask: Int){}

  override def glClear(mask: Int){}

  override def glEnableVertexAttribArray(index: Int){}

  override def glVertexAttrib1f(indx: Int, x: Float){}

  override def glUniformMatrix3fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer){}

  override def glUniformMatrix3fv(location: Int, count: Int, transpose: Boolean, value: Array[Float], offset: Int){}

  override def glUniform3f(location: Int, x: Float, y: Float, z: Float){}

  override def glGetShaderInfoLog(shader: Int): String = ""

  override def glClearStencil(s: Int){}

  override def glBindTexture(target: Int, texture: Int){}

  override def glStencilMaskSeparate(face: Int, mask: Int){}

  override def glGenFramebuffers(n: Int, framebuffers: IntBuffer){}

  override def glCopyTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, x: Int, y: Int, width: Int, height: Int){}

  override def glUniform3i(location: Int, x: Int, y: Int, z: Int){}

  override def glVertexAttrib4f(indx: Int, x: Float, y: Float, z: Float, w: Float){}

  override def glIsRenderbuffer(renderbuffer: Int): Boolean = true

  override def glGetTexParameterfv(target: Int, pname: Int, params: FloatBuffer){}

  override def glHint(target: Int, mode: Int){}

  override def glScissor(x: Int, y: Int, width: Int, height: Int){}

  override def glDeleteTexture(texture: Int){}

  override def glEnable(cap: Int){}

  override def glCullFace(mode: Int){}

  override def glGenBuffers(n: Int, buffers: IntBuffer){}

  override def glIsTexture(texture: Int): Boolean = true

  override def glGetIntegerv(pname: Int, params: IntBuffer){}

  override def glVertexAttrib3fv(indx: Int, values: FloatBuffer){}

  override def glIsFramebuffer(framebuffer: Int): Boolean = true

  override def glDepthRangef(zNear: Float, zFar: Float){}

  override def glGetFramebufferAttachmentParameteriv(target: Int, attachment: Int, pname: Int, params: IntBuffer){}

  override def glFlush(){}

  override def glDeleteRenderbuffers(n: Int, renderbuffers: IntBuffer){}

  override def glDeleteTextures(n: Int, textures: IntBuffer){}

  override def glCompileShader(shader: Int){}

  override def glStencilOp(fail: Int, zfail: Int, zpass: Int){}

  override def glBindAttribLocation(program: Int, index: Int, name: String){}

  override def glUniform2iv(location: Int, count: Int, v: IntBuffer){}

  override def glUniform2iv(location: Int, count: Int, v: Array[Int], offset: Int){}

  override def glViewport(x: Int, y: Int, width: Int, height: Int){}

  override def glDeleteFramebuffers(n: Int, framebuffers: IntBuffer){}

  override def glPixelStorei(pname: Int, param: Int){}

  override def glBindFramebuffer(target: Int, framebuffer: Int){}

  override def glBlendFunc(sfactor: Int, dfactor: Int){}

  override def glGetBooleanv(pname: Int, params: Buffer){}

  override def glGetError(): Int = 0

  override def glFramebufferTexture2D(target: Int, attachment: Int, textarget: Int, texture: Int, level: Int){}

  override def glShaderBinary(n: Int, shaders: IntBuffer, binaryformat: Int, binary: Buffer, length: Int){}

  override def glTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, width: Int, height: Int, format: Int, `type`: Int, pixels: Buffer){}

  override def glReleaseShaderCompiler(){}

  override def glTexImage2D(target: Int, level: Int, internalformat: Int, width: Int, height: Int, border: Int, format: Int, `type`: Int, pixels: Buffer){}

  override def glUniform2fv(location: Int, count: Int, v: FloatBuffer){}

  override def glUniform2fv(location: Int, count: Int, v: Array[Float], offset: Int){}

  override def glGetString(name: Int): String = ""

  override def glActiveTexture(texture: Int){}

  override def glGenTextures(n: Int, textures: IntBuffer){}

  override def glBufferSubData(target: Int, offset: Int, size: Int, data: Buffer){}

  override def glDeleteBuffers(n: Int, buffers: IntBuffer){}

  override def glVertexAttrib3f(indx: Int, x: Float, y: Float, z: Float){}

  override def glClearDepthf(depth: Float){}

  override def glGenFramebuffer(): Int = 0

  override def glDeleteShader(shader: Int){}

  override def glUniformMatrix2fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer){}

  override def glUniformMatrix2fv(location: Int, count: Int, transpose: Boolean, value: Array[Float], offset: Int){}

  override def glBufferData(target: Int, size: Int, data: Buffer, usage: Int){}

  override def glBlendColor(red: Float, green: Float, blue: Float, alpha: Float){}

  override def glReadPixels(x: Int, y: Int, width: Int, height: Int, format: Int, `type`: Int, pixels: Buffer){}

  override def glTexParameteriv(target: Int, pname: Int, params: IntBuffer){}

  override def glUniform2i(location: Int, x: Int, y: Int){}

  override def glGetVertexAttribPointerv(index: Int, pname: Int, pointer: Buffer){}

  override def glUniform4iv(location: Int, count: Int, v: IntBuffer){}

  override def glUniform4iv(location: Int, count: Int, v: Array[Int], offset: Int){}

  override def glGetBufferParameteriv(target: Int, pname: Int, params: IntBuffer){}

  override def glStencilMask(mask: Int){}

  override def glDeleteBuffer(buffer: Int){}

  override def glDrawElements(mode: Int, count: Int, `type`: Int, indices: Buffer){}

  override def glDrawElements(mode: Int, count: Int, `type`: Int, indices: Int){}

  override def glDeleteProgram(program: Int){}

  override def glBindBuffer(target: Int, buffer: Int){}

  override def glVertexAttrib2fv(indx: Int, values: FloatBuffer){}

  override def glFinish(){}

  override def glDrawArrays(mode: Int, first: Int, count: Int){}

  override def glRenderbufferStorage(target: Int, internalformat: Int, width: Int, height: Int){}

  override def glBindRenderbuffer(target: Int, renderbuffer: Int){}

  override def glIsProgram(program: Int): Boolean = true

  override def glGetUniformLocation(program: Int, name: String): Int = 0

  override def glUniform2f(location: Int, x: Float, y: Float){}

  override def glTexParameterfv(target: Int, pname: Int, params: FloatBuffer){}

  override def glLinkProgram(program: Int){}

  override def glGetActiveAttrib(program: Int, index: Int, size: IntBuffer, `type`: Buffer): String = ""

  override def glCompressedTexImage2D(target: Int, level: Int, internalformat: Int, width: Int, height: Int, border: Int, imageSize: Int, data: Buffer){}

  override def glBlendEquation(mode: Int){}

  override def glGenRenderbuffer(): Int = 0

  override def glGetProgramiv(program: Int, pname: Int, params: IntBuffer){}

  override def glUniform1iv(location: Int, count: Int, v: IntBuffer){}

  override def glUniform1iv(location: Int, count: Int, v: Array[Int], offset: Int){}

  override def glGetShaderiv(shader: Int, pname: Int, params: IntBuffer){}

  override def glUniform4fv(location: Int, count: Int, v: FloatBuffer){}

  override def glUniform4fv(location: Int, count: Int, v: Array[Float], offset: Int){}

  override def glAttachShader(program: Int, shader: Int){}

  override def glSampleCoverage(value: Float, invert: Boolean){}

  override def glColorMask(red: Boolean, green: Boolean, blue: Boolean, alpha: Boolean){}

  override def glUniformMatrix4fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer){}

  override def glUniformMatrix4fv(location: Int, count: Int, transpose: Boolean, value: Array[Float], offset: Int){}

  override def glGetProgramInfoLog(program: Int): String = ""

  override def glStencilOpSeparate(face: Int, fail: Int, zfail: Int, zpass: Int){}

  override def glIsEnabled(cap: Int): Boolean = true

  override def glGenTexture(): Int = 0

  override def glGetVertexAttribiv(index: Int, pname: Int, params: IntBuffer){}

  override def glVertexAttrib1fv(indx: Int, values: FloatBuffer){}

  override def glLineWidth(width: Float){}

  override def glDisable(cap: Int){}

  override def glUniform1fv(location: Int, count: Int, v: FloatBuffer){}

  override def glUniform1fv(location: Int, count: Int, v: Array[Float], offset: Int){}

  override def glIsShader(shader: Int): Boolean = true

  override def glDeleteFramebuffer(framebuffer: Int){}

  override def glDetachShader(program: Int, shader: Int){}

  override def glCopyTexImage2D(target: Int, level: Int, internalformat: Int, x: Int, y: Int, width: Int, height: Int, border: Int){}

  override def glUniform4f(location: Int, x: Float, y: Float, z: Float, w: Float){}

  override def glGenRenderbuffers(n: Int, renderbuffers: IntBuffer){}

  override def glCompressedTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, width: Int, height: Int, format: Int, imageSize: Int, data: Buffer){}

  override def glTexParameterf(target: Int, pname: Int, param: Float){}

  override def glVertexAttrib2f(indx: Int, x: Float, y: Float){}

  override def glBlendEquationSeparate(modeRGB: Int, modeAlpha: Int){}

  override def glUniform4i(location: Int, x: Int, y: Int, z: Int, w: Int){}

  override def glDeleteRenderbuffer(renderbuffer: Int){}

  override def glVertexAttrib4fv(indx: Int, values: FloatBuffer){}

  override def glClearColor(red: Float, green: Float, blue: Float, alpha: Float){}

  override def glGetRenderbufferParameteriv(target: Int, pname: Int, params: IntBuffer){}

  override def glCreateShader(`type`: Int): Int = 0

  override def glGetVertexAttribfv(index: Int, pname: Int, params: FloatBuffer){}

  override def glTexParameteri(target: Int, pname: Int, param: Int){}

  override def glPolygonOffset(factor: Float, units: Float){}

  override def glGetFloatv(pname: Int, params: FloatBuffer){}

  override def glFrontFace(mode: Int){}

  override def glStencilFunc(func: Int, ref: Int, mask: Int){}

  override def glShaderSource(shader: Int, string: String){}

  override def glUniform1f(location: Int, x: Float){}

  override def glVertexAttribPointer(indx: Int, size: Int, `type`: Int, normalized: Boolean, stride: Int, ptr: Buffer){}

  override def glVertexAttribPointer(indx: Int, size: Int, `type`: Int, normalized: Boolean, stride: Int, ptr: Int){}

  override def glCheckFramebufferStatus(target: Int): Int = 0

  override def glBlendFuncSeparate(srcRGB: Int, dstRGB: Int, srcAlpha: Int, dstAlpha: Int){}

  override def glGetUniformiv(program: Int, location: Int, params: IntBuffer){}

  override def glGetActiveUniform(program: Int, index: Int, size: IntBuffer, `type`: Buffer): String = ""

  override def glUniform3iv(location: Int, count: Int, v: IntBuffer){}

  override def glUniform3iv(location: Int, count: Int, v: Array[Int], offset: Int){}

  override def glDepthMask(flag: Boolean){}
}

File: src/test/kotlin/gameover/fwk/libgdx/FakeGL20.kt
package gameover.fwk.libgdx

import java.nio.FloatBuffer
import java.nio.Buffer
import java.nio.IntBuffer

import com.badlogic.gdx.graphics.GL20

/**
 * Fake GL for test.
 */
class FakeGL20 : GL20 {
    override fun glUniform3i(location: Int, x: Int, y: Int, z: Int) {}

    override fun glLineWidth(width: Float) {}

    override fun glDeleteShader(shader: Int) {}

    override fun glDetachShader(program: Int, shader: Int) {}

    override fun glVertexAttrib3f(indx: Int, x: Float, y: Float, z: Float) {}

    override fun glCompileShader(shader: Int) {}

    override fun glTexParameterfv(target: Int, pname: Int, params: FloatBuffer?) {}

    override fun glStencilFunc(func: Int, ref: Int, mask: Int) {}

    override fun glDeleteFramebuffer(framebuffer: Int) {}

    override fun glGenTexture(): Int {
        return 0
    }

    override fun glBindAttribLocation(program: Int, index: Int, name: String?) {}

    override fun glEnableVertexAttribArray(index: Int) {}

    override fun glReleaseShaderCompiler() {}

    override fun glUniform2f(location: Int, x: Float, y: Float) {}

    override fun glGetActiveAttrib(program: Int, index: Int, size: IntBuffer?, type: Buffer?): String {
        return ""
    }

    override fun glGenFramebuffer(): Int {
        return 0
    }

    override fun glUniformMatrix2fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer?) {}

    override fun glUniformMatrix2fv(location: Int, count: Int, transpose: Boolean, value: FloatArray?, offset: Int) {}

    override fun glUniform2fv(location: Int, count: Int, v: FloatBuffer?) {
    }

    override fun glUniform2fv(location: Int, count: Int, v: FloatArray?, offset: Int) {
    }

    override fun glUniform4iv(location: Int, count: Int, v: IntBuffer?) {
    }

    override fun glUniform4iv(location: Int, count: Int, v: IntArray?, offset: Int) {
    }

    override fun glColorMask(red: Boolean, green: Boolean, blue: Boolean, alpha: Boolean) {
    }

    override fun glPolygonOffset(factor: Float, units: Float) {
    }

    override fun glViewport(x: Int, y: Int, width: Int, height: Int) {
    }

    override fun glGetProgramiv(program: Int, pname: Int, params: IntBuffer?) {
    }

    override fun glGetBooleanv(pname: Int, params: Buffer?) {
    }

    override fun glGetBufferParameteriv(target: Int, pname: Int, params: IntBuffer?) {
    }

    override fun glDeleteTexture(texture: Int) {
    }

    override fun glGetVertexAttribiv(index: Int, pname: Int, params: IntBuffer?) {
    }

    override fun glVertexAttrib4fv(indx: Int, values: FloatBuffer?) {
        
    }

    override fun glTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, width: Int, height: Int, format: Int, type: Int, pixels: Buffer?) {
        
    }

    override fun glDeleteRenderbuffers(n: Int, renderbuffers: IntBuffer?) {
        
    }

    override fun glGetTexParameteriv(target: Int, pname: Int, params: IntBuffer?) {
        
    }

    override fun glGenTextures(n: Int, textures: IntBuffer?) {
        
    }

    override fun glStencilOpSeparate(face: Int, fail: Int, zfail: Int, zpass: Int) {
        
    }

    override fun glUniform2i(location: Int, x: Int, y: Int) {
        
    }

    override fun glCheckFramebufferStatus(target: Int): Int {
        return 0
    }

    override fun glDeleteTextures(n: Int, textures: IntBuffer?) {
        
    }

    override fun glBindRenderbuffer(target: Int, renderbuffer: Int) {
        
    }

    override fun glTexParameteriv(target: Int, pname: Int, params: IntBuffer?) {
        
    }

    override fun glVertexAttrib4f(indx: Int, x: Float, y: Float, z: Float, w: Float) {
        
    }

    override fun glDeleteBuffers(n: Int, buffers: IntBuffer?) {
        
    }

    override fun glGetProgramInfoLog(program: Int): String {
        return ""
    }

    override fun glIsRenderbuffer(renderbuffer: Int): Boolean {
        return false
    }

    override fun glFrontFace(mode: Int) {
        
    }

    override fun glUniform1iv(location: Int, count: Int, v: IntBuffer?) {
        
    }

    override fun glUniform1iv(location: Int, count: Int, v: IntArray?, offset: Int) {
        
    }

    override fun glClearDepthf(depth: Float) {
        
    }

    override fun glBindTexture(target: Int, texture: Int) {
        
    }

    override fun glGetUniformLocation(program: Int, name: String?): Int {
        return 0
    }

    override fun glPixelStorei(pname: Int, param: Int) {
        
    }

    override fun glHint(target: Int, mode: Int) {
        
    }

    override fun glFramebufferRenderbuffer(target: Int, attachment: Int, renderbuffertarget: Int, renderbuffer: Int) {
        
    }

    override fun glUniform1f(location: Int, x: Float) {
        
    }

    override fun glDepthMask(flag: Boolean) {
        
    }

    override fun glBlendColor(red: Float, green: Float, blue: Float, alpha: Float) {
        
    }

    override fun glUniformMatrix4fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer?) {
        
    }

    override fun glUniformMatrix4fv(location: Int, count: Int, transpose: Boolean, value: FloatArray?, offset: Int) {
        
    }

    override fun glBufferData(target: Int, size: Int, data: Buffer?, usage: Int) {
        
    }

    override fun glValidateProgram(program: Int) {
        
    }

    override fun glTexParameterf(target: Int, pname: Int, param: Float) {
        
    }

    override fun glIsFramebuffer(framebuffer: Int): Boolean {
        return false
    }

    override fun glDeleteBuffer(buffer: Int) {
        
    }

    override fun glShaderSource(shader: Int, string: String?) {
        
    }

    override fun glVertexAttrib2fv(indx: Int, values: FloatBuffer?) {
        
    }

    override fun glDeleteFramebuffers(n: Int, framebuffers: IntBuffer?) {
        
    }

    override fun glUniform4fv(location: Int, count: Int, v: FloatBuffer?) {
        
    }

    override fun glUniform4fv(location: Int, count: Int, v: FloatArray?, offset: Int) {
        
    }

    override fun glCompressedTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, width: Int, height: Int, format: Int, imageSize: Int, data: Buffer?) {
        
    }

    override fun glGenerateMipmap(target: Int) {
        
    }

    override fun glDeleteProgram(program: Int) {
        
    }

    override fun glFramebufferTexture2D(target: Int, attachment: Int, textarget: Int, texture: Int, level: Int) {
        
    }

    override fun glGenRenderbuffer(): Int {
        return 0
    }

    override fun glAttachShader(program: Int, shader: Int) {
        
    }

    override fun glBindBuffer(target: Int, buffer: Int) {
        
    }

    override fun glShaderBinary(n: Int, shaders: IntBuffer?, binaryformat: Int, binary: Buffer?, length: Int) {
        
    }

    override fun glDisable(cap: Int) {
        
    }

    override fun glGetRenderbufferParameteriv(target: Int, pname: Int, params: IntBuffer?) {
        
    }

    override fun glGetShaderInfoLog(shader: Int): String {
        return ""
    }

    override fun glGetActiveUniform(program: Int, index: Int, size: IntBuffer?, type: Buffer?): String {
        return ""
    }

    override fun glClearColor(red: Float, green: Float, blue: Float, alpha: Float) {
        
    }

    override fun glIsShader(shader: Int): Boolean {
        return false
    }

    override fun glUniform1i(location: Int, x: Int) {
        
    }

    override fun glBlendEquationSeparate(modeRGB: Int, modeAlpha: Int) {
        
    }

    override fun glScissor(x: Int, y: Int, width: Int, height: Int) {
        
    }

    override fun glCreateProgram(): Int {
        return 0
    }

    override fun glUniformMatrix3fv(location: Int, count: Int, transpose: Boolean, value: FloatBuffer?) {
        
    }

    override fun glUniformMatrix3fv(location: Int, count: Int, transpose: Boolean, value: FloatArray?, offset: Int) {
        
    }

    override fun glGetTexParameterfv(target: Int, pname: Int, params: FloatBuffer?) {
        
    }

    override fun glVertexAttrib1f(indx: Int, x: Float) {
        
    }

    override fun glUniform1fv(location: Int, count: Int, v: FloatBuffer?) {
        
    }

    override fun glUniform1fv(location: Int, count: Int, v: FloatArray?, offset: Int) {
        
    }

    override fun glUniform3iv(location: Int, count: Int, v: IntBuffer?) {
        
    }

    override fun glUniform3iv(location: Int, count: Int, v: IntArray?, offset: Int) {
        
    }

    override fun glTexImage2D(target: Int, level: Int, internalformat: Int, width: Int, height: Int, border: Int, format: Int, type: Int, pixels: Buffer?) {
        
    }

    override fun glVertexAttrib3fv(indx: Int, values: FloatBuffer?) {
        
    }

    override fun glBlendFunc(sfactor: Int, dfactor: Int) {
        
    }

    override fun glIsEnabled(cap: Int): Boolean {
        return false
    }

    override fun glGetAttribLocation(program: Int, name: String?): Int {
        return 0
    }

    override fun glDepthRangef(zNear: Float, zFar: Float) {
        
    }

    override fun glFlush() {
        
    }

    override fun glSampleCoverage(value: Float, invert: Boolean) {
        
    }

    override fun glCopyTexSubImage2D(target: Int, level: Int, xoffset: Int, yoffset: Int, x: Int, y: Int, width: Int, height: Int) {
        
    }

    override fun glGetShaderiv(shader: Int, pname: Int, params: IntBuffer?) {
        
    }

    override fun glGetUniformfv(program: Int, location: Int, params: FloatBuffer?) {
        
    }

    override fun glUniform4f(location: Int, x: Float, y: Float, z: Float, w: Float) {
        
    }

    override fun glClear(mask: Int) {
        
    }

    override fun glDepthFunc(func: Int) {
        
    }

    override fun glIsBuffer(buffer: Int): Boolean {
        return false
    }

    override fun glVertexAttribPointer(indx: Int, size: Int, type: Int, normalized: Boolean, stride: Int, ptr: Buffer?) {
        
    }

    override fun glVertexAttribPointer(indx: Int, size: Int, type: Int, normalized: Boolean, stride: Int, ptr: Int) {
        
    }

    override fun glStencilMaskSeparate(face: Int, mask: Int) {
        
    }

    override fun glDrawElements(mode: Int, count: Int, type: Int, indices: Buffer?) {
        
    }

    override fun glDrawElements(mode: Int, count: Int, type: Int, indices: Int) {
        
    }

    override fun glTexParameteri(target: Int, pname: Int, param: Int) {
        
    }

    override fun glUseProgram(program: Int) {
        
    }

    override fun glFinish() {
        
    }

    override fun glGetIntegerv(pname: Int, params: IntBuffer?) {
        
    }

    override fun glBlendEquation(mode: Int) {
        
    }

    override fun glUniform4i(location: Int, x: Int, y: Int, z: Int, w: Int) {
        
    }

    override fun glVertexAttrib1fv(indx: Int, values: FloatBuffer?) {
        
    }

    override fun glUniform3fv(location: Int, count: Int, v: FloatBuffer?) {
        
    }

    override fun glUniform3fv(location: Int, count: Int, v: FloatArray?, offset: Int) {
        
    }

    override fun glVertexAttrib2f(indx: Int, x: Float, y: Float) {
        
    }

    override fun glActiveTexture(texture: Int) {
        
    }

    override fun glCullFace(mode: Int) {
        
    }

    override fun glClearStencil(s: Int) {
        
    }

    override fun glGetFloatv(pname: Int, params: FloatBuffer?) {
        
    }

    override fun glDrawArrays(mode: Int, first: Int, count: Int) {
        
    }

    override fun glBindFramebuffer(target: Int, framebuffer: Int) {
        
    }

    override fun glGetError(): Int {
        return 0
    }

    override fun glBufferSubData(target: Int, offset: Int, size: Int, data: Buffer?) {
        
    }

    override fun glCopyTexImage2D(target: Int, level: Int, internalformat: Int, x: Int, y: Int, width: Int, height: Int, border: Int) {
        
    }

    override fun glIsProgram(program: Int): Boolean {
        return false
    }

    override fun glStencilOp(fail: Int, zfail: Int, zpass: Int) {
        
    }

    override fun glDisableVertexAttribArray(index: Int) {
        
    }

    override fun glGenBuffers(n: Int, buffers: IntBuffer?) {
        
    }

    override fun glGetAttachedShaders(program: Int, maxcount: Int, count: Buffer?, shaders: IntBuffer?) {
        
    }

    override fun glGenRenderbuffers(n: Int, renderbuffers: IntBuffer?) {
        
    }

    override fun glRenderbufferStorage(target: Int, internalformat: Int, width: Int, height: Int) {
        
    }

    override fun glUniform3f(location: Int, x: Float, y: Float, z: Float) {
        
    }

    override fun glReadPixels(x: Int, y: Int, width: Int, height: Int, format: Int, type: Int, pixels: Buffer?) {
        
    }

    override fun glStencilMask(mask: Int) {
        
    }

    override fun glBlendFuncSeparate(srcRGB: Int, dstRGB: Int, srcAlpha: Int, dstAlpha: Int) {
        
    }

    override fun glGetShaderPrecisionFormat(shadertype: Int, precisiontype: Int, range: IntBuffer?, precision: IntBuffer?) {
        
    }

    override fun glIsTexture(texture: Int): Boolean {
        return false
    }

    override fun glGetVertexAttribfv(index: Int, pname: Int, params: FloatBuffer?) {
        
    }

    override fun glGetVertexAttribPointerv(index: Int, pname: Int, pointer: Buffer?) {
        
    }

    override fun glCreateShader(type: Int): Int {
        return 0
    }

    override fun glStencilFuncSeparate(face: Int, func: Int, ref: Int, mask: Int) {
        
    }

    override fun glGetString(name: Int): String {
        return ""
    }

    override fun glCompressedTexImage2D(target: Int, level: Int, internalformat: Int, width: Int, height: Int, border: Int, imageSize: Int, data: Buffer?) {
        
    }

    override fun glUniform2iv(location: Int, count: Int, v: IntBuffer?) {
        
    }

    override fun glUniform2iv(location: Int, count: Int, v: IntArray?, offset: Int) {
        
    }

    override fun glGenBuffer(): Int {
        return 0
    }

    override fun glEnable(cap: Int) {
        
    }

    override fun glGetUniformiv(program: Int, location: Int, params: IntBuffer?) {
        
    }

    override fun glGetFramebufferAttachmentParameteriv(target: Int, attachment: Int, pname: Int, params: IntBuffer?) {
        
    }

    override fun glDeleteRenderbuffer(renderbuffer: Int) {
        
    }

    override fun glGenFramebuffers(n: Int, framebuffers: IntBuffer?) {
        
    }

    override fun glLinkProgram(program: Int) {
        
    }

}